### PR TITLE
Specifying Kafka version as 1.0.0 in Note for Kafka with EH

### PR DESCRIPTION
**Pull request by DeepanshuA**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Issue reference
Please reference the issue this PR will close: https://github.com/dapr/components-contrib/issues/2874

### CODE REVIEW

File: daprdocs/content/en/reference/components-reference/supported-bindings/kafka.md [hunk 1]
```diff
@@ -67,6 +67,10 @@ spec:
 | oidcClientID | N | Input/Output | The OAuth2 client ID that has been provisioned in the identity provider. Required when `authType` is set to `oidc` | `dapr-kafka` |
 | oidcClientSecret | N | Input/Output | The OAuth2 client secret that has been provisioned in the identity provider: Required when `authType` is set to `oidc` | `"KeFg23!"` |
 | oidcScopes | N | Input/Output | Comma-delimited list of OAuth2/OIDC scopes to request with the access token. Recommended when `authType` is set to `oidc`. Defaults to `"openid"` | `"openid,kafka-prod"` |
+| version | N | Input/Output | Kafka cluster version. Defaults to 2.0.0. Please note that this needs to be mandatorily set to `1.0.0` for EventHubs with Kafka. | `1.0.0` |
+
+#### Note
+Metadata `version` needs to be mandatorily set to `1.0.0` for anyone using EventHubs with Kafka.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/kafka.md]:
```suggestion
The metadata `version` must be set to `1.0.0` when using Azure EventHubs with Kafka.
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md [hunk 2]
```diff
@@ -65,7 +65,7 @@ spec:
 | maxMessageBytes     | N | The maximum size in bytes allowed for a single Kafka message. Defaults to 1024. | `2048`
 | consumeRetryInterval | N | The interval between retries when attempting to consume topics. Treats numbers without suffix as milliseconds. Defaults to 100ms. | `200ms` |
 | consumeRetryEnabled | N | Disable consume retry by setting `"false"` | `"true"`, `"false"` |
-| version               | N | Kafka cluster version. Defaults to 2.0.0.0 | `0.10.2.0` |
+| version               | N | Kafka cluster version. Defaults to 2.0.0.0. Please note that this needs to be mandatorily set to `1.0.0` for EventHubs with Kafka. | `0.10.2.0` |
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md]:
```suggestion
| version               | N | Kafka cluster version. Defaults to 2.0.0.0. Note that this must be set to `1.0.0` if you using Azure EventHubs with Kafka. | `0.10.2.0` |
```

Comment (msfussell) [reply daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md]: Should this be `1.0.0.0` ?

File: daprdocs/content/en/reference/components-reference/supported-bindings/kafka.md (hunk 1 above)
Comment (msfussell) [reply daprdocs/content/en/reference/components-reference/supported-bindings/kafka.md]: Should this be `1.0.0.0` ?

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md [hunk 3]
```diff
@@ -78,6 +78,9 @@ spec:
 
 The `secretKeyRef` above is referencing  a [kubernetes secrets store]({{< ref kubernetes-secret-store.md >}}) to access the tls information. Visit [here]({{< ref setup-secret-store.md >}}) to learn more about how to configure a secret store component.
 
+#### Note
+Metadata `version` needs to be mandatorily set to `1.0.0` for anyone using EventHubs with Kafka.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md]:
```suggestion
The metadata `version` must be set to `1.0.0` when using Azure EventHubs with Kafka.
```

**Comment by msfussell** [comment]
@DeepanshuA - Left you some feedback

**Review by mukundansundar — COMMENTED** [review]
Can you add a reference in Azure EventHubs component as well?

File: daprdocs/content/en/reference/components-reference/supported-bindings/kafka.md (hunk 1 above)
Comment (DeepanshuA) [reply daprdocs/content/en/reference/components-reference/supported-bindings/kafka.md]: Actually, before `1.0.0`, versions had 4 parts, like `0.10.2.0` or `0.11.0.3`
After and from `1.0.0`, it has 3 parts.

Ref: https://kafka.apache.org/downloads#1.0.0

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md (hunk 2 above)
Comment (DeepanshuA) [reply daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md]: No, `1.0.0` is correct. Infact, changed `2.0.0.0` to `2.0.0`

**Comment by DeepanshuA** [comment]
> Can you add a reference in Azure EventHubs component as well?

Correct. We had a conversation on the same lines.
Then, I had a conversation with @berndverst and he suggested that EH doc should not be updated for that.
In this particular case, I don't have a very strong inclination for any side, but yeah - I don't see a harm in putting some Note in EH docs for this.
@berndverst @mukundansundar Please let me know if we want to put something in EH docs or not.

**Comment by msfussell** [comment]
@mukundansundar - Are you good here with this PR? Do you want to add any changes? Would like to get this merged.

**Comment by mukundansundar** [comment]
> @mukundansundar - Are you good here with this PR? Do you want to add any changes? Would like to get this merged.

From a user perspective if we can add a back link from EH docs to Kafka pointing that EH can be used with Kafka component, it would be good. 
This PR can be merged though.

**Review by msfussell — APPROVED** [review]
Approved

**Pull request merged**